src/htmlParser.js

```diff
@@ -261,7 +261,7 @@ export default class HtmlParser {
     }
 
     if (options.fix) {
-      this._fixedReadToken = fixedReadTokenFactory(this, options);
+      this._fixedReadToken = fixedReadTokenFactory(this, options, () => {return this._readTokenImpl();});
```

Comment:
Any reason `() => this._readTokenImpl()` wouldn't work?  Or even just `this._readTokenImpl`?